src/commandHandlers/roles.ts

```diff
@@ -1,35 +1,94 @@
 import { MessageEmbed } from 'discord.js';
 
 import { getRoles } from './guild.service';
-import config from '../config';
+import config, { selfAssignableRoles } from '../config';
 
 /**
  * This command lists the available roles on the discord server
  */
 export const command = async (arg: [string, string], embed: MessageEmbed) => {
   const roles = await getRoles();
-  const rolesList = roles
-    .filter((r) => !r.name.includes('everyone')) // Filter the default role everyone has access to
-    .map((discordRole) => {
-      const role = Object.values(config.ROLE).find(
-        (r) => r.name === discordRole.name
-      );
-      const roleDescription = role ? `- ${role.description}` : '';
-
-      return `\n• ${discordRole.toString()} ${roleDescription}`;
-    });
+  const specificRole = arg[1];
 
-  embed.setTitle('Available Roles')
-    .setDescription(`Here is the list of roles on this server. You can assign almost any role to yourself. Some of the roles are admin only or given to you via a condition!
-        ${rolesList}
+  if (specificRole) {
+    if (arg[1] === '-a' || arg[1] === 'all') {
+      const describedRoles: string[] = [];
+      const rolesList: string[] = [];
+      roles.forEach((role) => {
+        if (role.name.includes('everyone')) {
+          return;
+        }
+        const roleConfig = Object.values(config.ROLE).find(
+          (r) => r.name === role.name
+        );
+        if (roleConfig && roleConfig.description) {
+          describedRoles.push(
+            `\n${role.toString()} - ${roleConfig.description}`
+          );
+        } else {
+          rolesList.push(` ${role.toString()}`);
+        }
+      });
```

Comment:
I agree with @eddiejaoude in separating the logic on the nested `if` statements, in order to get smaller functions that are easier to reason about.

This block of code serves one purpose which is to _get the role lists that are displayed to the user_. The differences between this block and the one in "Assignable-roles" option are:
- The new line (`\n`) is placed differently: ``\n${role.toString()} - ${roleConfig.description}`` and ``${role.toString()} - ${roleConfig.description}\n``
- The filter used to ignore roles: `role.name.includes('everyone')` and `!selfAssignableRoles.includes(role.name)`

Is there a noticeable difference between having the `\n` at the start or at the end of the string in the Discord message @Vyvy-vi? If there isn't we could choose one and move this logic to a separate function. Something like this:
```ts
// Returns two lists, one with all the role names and the other with role names followed by their description.
function getRoleLists(roles: Collection<string, Role>, isValidRole: (role: Role) => boolean): [string[], string[]]{
  const describedRoles: string[] = [];
  const rolesList: string[] = [];
  roles.forEach((role) => {
    if (!isValidRole(role)) {
      return;
    }
    const roleConfig = Object.values(config.ROLE).find(
      (r) => r.name === role.name
    );
    if (roleConfig && roleConfig.description) {
      describedRoles.push(
        `\n${role.toString()} - ${roleConfig.description}`
      );
    } else {
      rolesList.push(` ${role.toString()}`);
    }
  });
  return [rolesList, describedRoles];
}
```

What do you think @Vyvy-vi ?

Reply:
yes, shifting the logic to the function would make things waaay more better :+1:

> Is there a noticeable difference between having the \n at the start or at the end of the string in the Discord message @Vyvy-vi? If there isn't we could choose one and move this logic to a separate function.

There isn't much difference.. the `\n` is there for formatting reasons, which could be adjusted later on, when putting the returned lists in the embed(we could use `join` rather than having the newline escape sequences)